### PR TITLE
renamed 'pbc' kwarg to 'wrap' for Group methods

### PR DESCRIPTION
**Pull request by richardjgowers**
Related to issue #1760 in particular [@orbeckst ](https://github.com/MDAnalysis/mdanalysis/issues/1760#issuecomment-360320659) here

Changes made in this Pull Request:
 - pbc kwarg changed to wrap for Group methods

The `pbc` kwarg for center_of_mass has been confusing people for a while, it's not clear if this wraps (ie forcing atoms to inside the unit cell) or unwrap (ie makes bonds whole, possibly putting atoms outside the unit cell). 

The `pbc` option will still work for now but raise a deprecation warning

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/core/groups.py
Comment (richardjgowers) [inline package/MDAnalysis/core/groups.py:394]: Here we used to allow `pbc` and `unwrap` if compound wasn't group, this slipped through review here: https://github.com/MDAnalysis/mdanalysis/pull/2275/files#r296183096

For sanity's sake, I'm saying that if you want a combination of wrapping/unwrapping you need to do multiple calls

**Comment by richardjgowers** [comment]
I'm also not sure what we should target for the pbc deprecation.  It's not overly urgent we just need to nudge people away from using `center_of_mass(pbc=True)` but it's not going to break anything to leave the undocumented option

**Comment by jbarnoud** [comment]
Just throwing #2299 here for reference.

File: package/MDAnalysis/core/groups.py
Comment (jbarnoud) [inline package/MDAnalysis/core/groups.py:394]: The tone sounds a bit harsh, maybe.

**Comment by orbeckst** [comment]
If you split the PR into two:
1. deprecations (against master), to go into 1.0.1
2. this PR where you don't bother with deprecations

**Comment by richardjgowers** [comment]
@orbeckst the deprecations are in this PR, `pbc=True` still works but issues a warning.  This doesn't need backporting.

**Comment by orbeckst** [comment]
I thought you wanted to get rid of pbc=True for 2.0. You can if you put the deprecations into 1.0.1.

Otherwise the next time you can get rid of them is 3.0.

**Comment by IAlibay** [comment]
What's the status of this? I'm making a finalised list of the 2.0 milestones. I've included it for now.

**Comment by orbeckst** [comment]
I think my review comment still stands: if anything is removed in 2.0, it needs to be deprecated in 1.x

**Comment by orbeckst** [comment]
My opinion remains that 
- the *pbc* kwarg for should be deprecated in 1.1 (and we allow *wrap* as a synonym as in this PR)
- we replace *pbc* with *wrap* in 2.0

Alternatively, we go ahead with the PR as it stands and allow *wrap* and *pbc* as synonyms for 2.x. 

Given that @richardjgowers and I seem to be of different opinion how to move forward, it would be useful to have some other devs offer opinions.

**Comment by IAlibay** [comment]
> My opinion remains that
> 
> * the _pbc_ kwarg for should be deprecated in 1.1 (and we allow _wrap_ as a synonym as in this PR)
> * we replace _pbc_ with _wrap_ in 2.0
> 
> Alternatively, we go ahead with the PR as it stands and allow _wrap_ and _pbc_ as synonyms for 2.x.
> 
> Given that @richardjgowers and I seem to be of different opinion how to move forward, it would be useful to have some other devs offer opinions.

Whilst I agree that deprecating in 1.1 would be cleaner. I think at this point we just need to complete #3242 and cut a release. Probably going with synonyms and then deprecating so that we can remove in 3.0 would be our best option.

File: package/MDAnalysis/core/groups.py
Comment (orbeckst) [inline package/MDAnalysis/core/groups.py:394]: add "and will be removed in 3.0"

File: package/MDAnalysis/core/groups.py
Comment (orbeckst) [inline package/MDAnalysis/core/groups.py:806]: add deprecation note here, too, — `pbc` is still accepted as a synonym for `wrap` but will be removed in 3.0

File: package/MDAnalysis/core/groups.py
Comment (orbeckst) [inline package/MDAnalysis/core/groups.py:941]: deprecation note as above

File: package/MDAnalysis/core/groups.py
Comment (orbeckst) [reply package/MDAnalysis/core/groups.py:394]: I added comments below that the docs need a deprecation notice in the docs. Maybe you can teach the wrapper to add it automatically?

File: package/MDAnalysis/core/groups.py
Comment (orbeckst) [inline package/MDAnalysis/core/groups.py:1115]: add deprecation note to docs

File: package/MDAnalysis/core/groups.py
Comment (orbeckst) [inline package/MDAnalysis/core/groups.py:1150]: add deprecation note to docs

File: package/MDAnalysis/core/topologyattrs.py
Comment (orbeckst) [inline package/MDAnalysis/core/topologyattrs.py:1153]: add deprecation note to docs

File: package/MDAnalysis/core/topologyattrs.py
Comment (orbeckst) [inline package/MDAnalysis/core/topologyattrs.py:1214]: add deprecation note to docs

File: package/MDAnalysis/core/topologyattrs.py
Comment (orbeckst) [inline package/MDAnalysis/core/topologyattrs.py:1207]: Could you please document the `unwrap` and `compound` parameters that were added in 0.20.0? I know, it's not your fault. But at the moment the docs are confusing, especially as both are not even explicit (and hidden as kwarg).  Making them explicit in the signature would also be good.

File: package/MDAnalysis/core/topologyattrs.py
Comment (orbeckst) [inline package/MDAnalysis/core/topologyattrs.py:1269]: Could you please document the `unwrap` and `compound` parameters that were added in 0.20.0? I know, it's not your fault. But at the moment the docs are confusing, especially as both are not even explicit (and hidden as kwarg).  Making them explicit in the signature would also be good.

File: package/MDAnalysis/core/topologyattrs.py
Comment (orbeckst) [inline package/MDAnalysis/core/topologyattrs.py:1275]: deprecation note

File: package/MDAnalysis/core/topologyattrs.py
Comment (orbeckst) [inline package/MDAnalysis/core/topologyattrs.py:1320]: deprecation note

File: package/MDAnalysis/core/topologyattrs.py
Comment (orbeckst) [inline package/MDAnalysis/core/topologyattrs.py:1297]: Can we get rid of kwargs here? They are not used.

Comment (orbeckst) [reply package/MDAnalysis/core/topologyattrs.py:1297]: Should this function also be able to use `unwrap`? (If yes, it can be a new issue.)

File: package/MDAnalysis/core/topologyattrs.py
Comment (orbeckst) [inline package/MDAnalysis/core/topologyattrs.py:1356]: versionchanged and deprecation note are missing here

File: package/MDAnalysis/core/topologyattrs.py
Comment (orbeckst) [inline package/MDAnalysis/core/topologyattrs.py:1413]: versionchanged and deprecation note missing

Comment (orbeckst) [reply package/MDAnalysis/core/topologyattrs.py:1413]: Should this function be able to use `unwrap` and `compound`? If so, raise issue.

**Review by orbeckst — CHANGES_REQUESTED** [review]
I am swayed by @IAlibay, let's deprecate `pbc` for removal in 3.0.

- documentation
- please also fix the benchmarks in `benchmarks/benchmarks/ag_methods.py`

**Comment by orbeckst** [comment]
@IAlibay this one was not on 2.0.0 milestone. It does not have to be because we'll deprecate for 3.0 but because unwrapping is something that we might want for the workshop (?), I thought I mention it.

**Comment by IAlibay** [comment]
@orbeckst are we good moving this to the 2.1 milestone?

**Comment by orbeckst** [comment]
sure, move it

**Comment by orbeckst** [comment]
Going through the comments, there doesn't seem to be an enormous amount of work to do, except that all the conflicts need to be resolved.

**Comment by jbarnoud** [comment]
@richardjgowers do you want to give the final touches on that PR? If you don't have time, I should be able to take care of it next week.

**Comment by jbarnoud** [comment]
I rebased the changes and fixed the conflicts. The tests are broken, though, because some tests have been introduced that rely on both `unwrap` and `warp` being used on the same call. I need to fix that. Some other tests may need fixing.

TODO:
* [x] fix broken tests
* [x] change version in the deprecation notices
* [x] mention version 3.0 in the deprecation warning

**Comment by richardjgowers** [comment]
@jbarnoud yes there are some things which have both unwrap and wrap.  I think the idea was to first unwrap to calculate e.g. a CoM, then wrap these results back inside.  I think this is extremely confusing as it's not clear which of wrap/unwrap is first see https://github.com/MDAnalysis/mdanalysis/pull/2857#discussion_r456911683

**Comment by jbarnoud** [comment]
@richardjgowers Indeed. Removing ambiguity is a good thing. My main issue with the broken tests is that it is not clear if they are meant to test the wrap and the unwrap.

**Comment by jbarnoud** [comment]
@orbeckst Assuming the tests pass, it should be good to go. Since there is a deprecation, it would be good to have it in 2.1.0 that @IAlibay hopes to push soon.

File: package/MDAnalysis/core/topologyattrs.py
Comment (orbeckst) [reply package/MDAnalysis/core/topologyattrs.py:1567]: Can you add the missing docs here or should I raise a new issue?

File: package/MDAnalysis/core/topologyattrs.py
Comment (orbeckst) [reply package/MDAnalysis/core/topologyattrs.py:1674]: Not applicable. Ignore above comment.

**Review by orbeckst — CHANGES_REQUESTED** [review]
I had a quick look and as far as I can tell there are still a few minor documentation issues left.

**Comment by orbeckst** [comment]
I am doing the doc updates.

**Review by orbeckst — APPROVED** [review]

**Comment by orbeckst** [comment]
I hope the tests pass now. I am approving now because I might not be able to get to it later during the day and in principle at least everything is there. Another pair of 👀 would be good before merging.

**Review by richardjgowers — COMMENTED** [review]
Lgtm!

**Pull request merged**
